MaxText/decode.py

```diff
@@ -70,13 +71,13 @@ def encode_strings(strs, max_len, tokenizer, mesh):
 
   for i, s in enumerate(strs):
     toks = tokenizer.tokenize(s).numpy()
-    assert toks.shape[0] < max_len, f"We aren't able to tokenize input {i}, it is too long"
+    assert toks.shape[0]-1 <= max_len, f"We aren't able to tokenize input {i}, it is too long"
```

Comment:
I hadn't understood what you meant by off by two. Are we sure we still want to remove EOS here? (Is it because we aren't setting config.use_eos to false?

Reply:
right, we are leaving `config.use_eos=True` like it was previously. Please let me know if you think that's not the right thing.

Reply:
I think you're saying for inference we should set config.use_eos=false?

Reply:
done - thanks!